cmp/tab/TabSwitcher.js

```diff
@@ -17,23 +17,32 @@ import {omit} from 'lodash';
  * This TabSwitcher controls the selected tab of a TabContainer via a shared TabContainerModel,
  * which is expected to be provided to this TabSwitcher.
  *
- * The switcherPosition configuration on the TabContainerModel controls how this switcher will be
- * rendered. For 'top' or 'bottom' switcherPositions this switcher will be rendered in horizontal
- * mode. For 'left' or 'right' switcher positions this switcher will be rendered in vertical mode.
+ * The orientation property controls how this switcher will be rendered. For 'top' or 'bottom'
+ * orientations this switcher will be rendered in horizontal mode. For 'left' or 'right'
+ * orientations this switcher will be rendered in vertical mode.
  *
  * @see TabContainerModel
  */
 @HoistComponent()
 export class TabSwitcher extends Component {
     static propTypes = {
         /** Model to switch tabs on. Should be shared with a TabContainer. */
-        model: PT.instanceOf(TabContainerModel)
+        model: PT.instanceOf(TabContainerModel),
+        /** The position relative to the TabContainer this switcher is controlling. Defaults to 'top'. */
+        orientation: PT.oneOf(['top', 'bottom', 'left', 'right'])
+    };
+
+    static defaultProps = {
+        orientation: 'top'
```

Comment:
We haven't been using defaultProps in general, because they are not flexible enough to handle alot of the dynamic defaults that we do.  Instead we have been generally implementing the defaults in code in the render method.  We should discuss this further -- obviously what you have here is pretty succinct

Reply:
Agreed good topic for discussion - I have a branch with a bunch of HoistField class related standardizations / cleanups that I need to get back to.  There we had some pretty heavy use of defaultProps and IIRC I continued in that direction....

Reply:
I saw defaultProps used elsewhere and thought it made sense here. The non-react-component parallel would be initializing class member variables when declaring, which I would still do even if some members needed to be initialized dynamically in the ctor. I would also raise the question of whether render is the best place to handle dynamic defaults instead of the ctor, unless the default value needs to be dynamic in such a way where if it wasn't explicitly provided through props, it is dependent on another prop which may change after the component is instantiated (though that smells like an anti-pattern)